### PR TITLE
Issue #950, part 1: Extended CIMParameter to represent parameter values

### PR DESCRIPTION
**Pull request by andy-maier**
This is part 1, which just extends CIMParameter as needed. Part 2 will be separate and will exploit that in InvokeMethod.
For details, see the commit message.
Ready for review and merge.

### CODE REVIEW

File: docs/changes.rst [hunk 1]
```diff
@@ -218,6 +218,13 @@ Enhancements
 * Added support for retrieving the operation recorders of a connection
   via a new `operation_recorders` read-only property (Issue #976).
 
+* Extended `CIMParameter` to represent CIM parameter values in method
+  invocations. As part of that, undeprecated its `value` property and
```
Comment (KSchopmeyer) [inline docs/changes.rst]: change undeprecated to removed deprecation from

File: pywbem/cim_obj.py [hunk 2]
```diff
@@ -5211,7 +5213,25 @@ def tomof(self, indent=0, maxline=MAX_MOF_LINE):
...
+      * The `value` attribute is the actual value of the parameter.
+      * Qualifiers are not allowed.
+
+    For parameter declarations in method declarations:
+
+      * The `value` attribute is ignored.
+      * Qualifiers are allowed.
+
+    Scalar (=non-array) parameters and items in array parameters may have a
+    value of NULL (= `None`), any primitive CIM data type, reference type, and
```
Comment (KSchopmeyer) [inline pywbem/cim_obj.py]: Should last word in sentence be or rather than any?

File: pywbem/cim_obj.py [hunk 3]
```diff
@@ -5298,13 +5320,23 @@ def __init__(self, name, type, reference_class=None, is_array=None,
...
+
+          embedded_object (:term:`string`):
+            A string value indicating the kind of embedded object represented
+            by the parameter value. Has no meaning for parameter declarations.
+
+            For details about the possible values, see the corresponding
+            attribute.
+
+            `None` means that the value is unspecified, causing the same-named
+            attribute in the ``CIMParameter`` object to be inferred. An
```
Comment (KSchopmeyer) [inline pywbem/cim_obj.py]: I would guess that the reader would ask at this point what we mean by 'inferred'

File: pywbem/cim_obj.py (hunk 2 above)
Comment (KSchopmeyer) [reply pywbem/cim_obj.py]: clarify comment.  Should last work on line be or rather than and

File: docs/changes.rst (hunk 1 above)
Comment (andy-maier) [reply docs/changes.rst]: Done.

File: pywbem/cim_obj.py (hunk 2 above)
Comment (andy-maier) [reply pywbem/cim_obj.py]: Done.

File: pywbem/cim_obj.py (hunk 3 above)
Comment (andy-maier) [reply pywbem/cim_obj.py]: DONE - clarified that it is inferred from the parameter value.

**Pull request merged**